src/feedback/FeedbackForm.tsx

```diff
@@ -56,53 +68,125 @@ export const FeedbackForm: FunctionComponent<FeedbackFormProps> = ({
     }
   };
 
-  if (status === "submitted") {
-    return (
-      <div class="submitted">
-        <p class="icon">🙏</p>
-        <p class="text">{t.SuccessMessage}</p>
-      </div>
-    );
-  }
+  const containerRef = useRef<HTMLDivElement>(null);
+  const formRef = useRef<HTMLFormElement>(null);
+  const headerRef = useRef<HTMLDivElement>(null);
+  const expandedContentRef = useRef<HTMLDivElement>(null);
+  const submittedRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    if (containerRef.current === null) return;
+    if (formRef.current === null) return;
+    if (headerRef.current === null) return;
+    if (expandedContentRef.current === null) return;
+    if (submittedRef.current === null) return;
+
+    if (status === "submitted") {
+      formRef.current.style.opacity = "0";
+      formRef.current.style.pointerEvents = "none";
+      containerRef.current.style.maxHeight =
+        submittedRef.current.clientHeight + "px";
+
+      // Fade in "submitted" step once container has resized
+      setTimeout(() => {
+        submittedRef.current!.style.position = "relative";
+        submittedRef.current!.style.opacity = "1";
+        submittedRef.current!.style.pointerEvents = "all";
+        setShowForm(false);
+      }, ANIMATION_SPEED + 10);
+    } else {
+      const isExpanded = openWithCommentVisible || hasRating;
+
+      containerRef.current.style.maxHeight = isExpanded
+        ? "400px" // TODO: reconsider?
```

Comment:
Unfortunately, the way `maxHeight` and transitions work means that the size just gets clamped and the transition won't end at the actual height but instead the maxHeight. The result being that the transition stops abruptly. 
Perhaps there's another way we can achieve this resizing while still take dynamic question text height into account?

Reply:
Definitely, this has been on my list to sort out. I'm currently calculating the exact size of the initial screen (before it expands to show the comment) and of the success screen, so was planning on using the same trick here for the full expanded screen. i.e. setting max-height to (score div height + comment div height + gap height) should work 🤞